mule-extensions-api/src/main/java/org/mule/runtime/extension/internal/dsl/syntax/XmlDslSyntaxResolver.java

```diff
@@ -214,32 +217,31 @@ public void visitArrayType(ArrayType arrayType) {
                                  @Override
                                  public void visitObject(ObjectType objectType) {
                                    addAttributeName(builder, parameter, isContent, dslConfig);
-                                   builder.withNamespace(namespace.get(), namespaceUri.get())
-                                       .withElementName(elementName.get());
-
-                                   resolveObjectDsl(objectType, builder, isContent, dslConfig, expressionSupport);
-                                 }
-
-                                 @Override
-                                 public void visitDictionary(DictionaryType dictionaryType) {
-                                   addAttributeName(builder, parameter, isContent, dslConfig);
+                                   if (objectType.isOpen()) {
```

Comment:
why does this only apply to open types? A closed type should also allow content parameters , inline declaration, etc.